.github/workflows/newman.yml

```diff
@@ -41,8 +48,179 @@ jobs:
           echo "GITHUB_OUTPUT contents:"
           cat $GITHUB_OUTPUT
 
+  associate-test-cases:
+      runs-on: ubuntu-22.04
+      outputs:
+        test_exec: ${{ steps.xray-newman.outputs.test_exec }}
+
+      defaults:
+       run:
+        shell: bash
+        working-directory: centreon/tests/rest_api
+
+      steps:
+        - name: Checkout Repository
+          uses: actions/checkout@v4
+
+        - name: Generate Xray Token
+          id: generate-xray-token
+          run: |
+                  token_response=$(curl -H "Content-Type: application/json" -X POST --data "{\"client_id\": \"${{ secrets.CLIENT_ID }}\", \"client_secret\": \"${{ secrets.CLIENT_SECRET }}\"}" "https://xray.cloud.getxray.app/api/v1/authenticate")
+                  xray_token=$(echo "$token_response" | sed -n 's/.*"\(.*\)".*/\1/p')
+                  echo "this is the xray token : $xray_token"
+                  echo "xray_token=$xray_token" >> $GITHUB_OUTPUT
+          shell: bash
+
+        - name: Associate Test Cases with Test plan
+          id: xray-newman
+          run: |
+            xray_graphql_getTestPlan='{
+              "query": "query GetTestPlan($issueId: String) { getTestPlan(issueId: $issueId) { issueId tests(limit: 100) { results { issueId testType { name } } } } }",
+              "variables": {
+                "issueId": "73560"

```

Comment:
wait current @a-eljazouly work to get this value dynamically or create it if it does not exist
@a-eljazouly maybe do a separated action which generate an output `testPlanId` to reuse it in this workflow

Reply:
@MedWassCh just use an other id regarding environments